### DISCUSSION TITLE
[ Implemented ] Configure hiding meshes when entering sketch environment

### DISCUSSION THREAD

**Discussion by Cheaterman** [hlorus/CAD_Sketcher#113] [Feature Requests]
**Motivation / Existing Problem**
Currently the behavior is slightly inconsistent, meshes are not hidden when creating a new sketch but they are when editing an existing sketch.

**Description**
On Discord, I suggested we simply make that behavior consistent. @TheMakerTales suggested we go the extra mile and make this a configurable behavior, which does sound like a great idea!

**Alternative Solutions**
Otherwise, as initially suggested, we could simply make the behavior consistent - either hide or not in both cases.

**Mockup**
No real need, although I guess it would be a checkbox in the plugin options?

- [ ] Hide meshes when entering sketch environment

**Comment by paynterf** [comment]
As a new user on both Blender and CAD Sketcher, trying to follow Jonathon's new 0.24 YT video, I've been bitten repeatedly by this.  Jonathon's video clearly shows that meshes stay visible when a new sketch is added, but on my system, the previous mesh(s) all disappear when the 'Add Sketch' button is pressed.  Took me forever to figure out how to make them visible again

**Reply by linc01** [reply]
You can use the "view object types" to choose visible/invisible types.

**Comment by TheMakerTales** [comment]
This has now been implemented.
Please check the CAD Sketcher preferences and you'll see a little tick box there to set your preference.
